cmd/krel/cmd/testgrid.go

```diff
@@ -0,0 +1,269 @@
+	DashboardName string `mapstructure:"dashboard_name"`
+}
+
+type TestGridJob struct {
+	DashboardName string
+	JobName       string
+	Status        string
+	GCSLocation   string
+}
+
+const (
+	statePassing = "PASSING"
+	stateFlaky   = "FLAKY"
+	stateFailing = "FAILING"
+
+	boardInforming = "informing"
+	boardBlocking  = "blocking"
+
+	layoutISO = "2006-01-02"
+)
+
+// ffCmd represents the base command when called without any subcommands
+var testGridCmd = &cobra.Command{
+	Use:           "testgridshot --branch <release-branch>",
+	Short:         "Take a screenshot of the testgrid dashboards",
+	Example:       "krel testgridshot --branch 1.17",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runTestGridShot(testGridOpts)
+	},
+}
+
+func init() {
+	testGridCmd.PersistentFlags().StringVar(&testGridOpts.branch, "branch",
+		"master", "From which release branch will get the testgrid dashboard")
+
+	testGridCmd.PersistentFlags().StringSliceVar(&testGridOpts.boards, "boards", []string{boardBlocking, boardInforming},
+		"Which Boards to retrieve the dashboards, defaults to blocking and informing")
+
+	testGridCmd.PersistentFlags().StringSliceVar(&testGridOpts.states, "states", []string{stateFailing},
+		"Which States to watch for each dashboard, default to failing")
+
+	testGridCmd.PersistentFlags().StringVar(&testGridOpts.testgridURL,
+		"testgrid-url", "https://testgrid.k8s.io", "The TestGrid URL")
+
+	testGridCmd.PersistentFlags().StringVar(&testGridOpts.renderTronURL,
+		"rendertron-url", "https://render-tron.appspot.com/screenshot", "The RenderTron URL service")
+
+	testGridCmd.PersistentFlags().StringVar(&testGridOpts.bucket, "bucket", "k8s-staging-releng",
+		"The name of the bucket to upload the images to. The files will be put into '/testgridshot/<release>/<datetime>_<rand>/...'. Defaults to k8s-staging-releng")
+
+	rootCmd.AddCommand(testGridCmd)
+}
+
+func runTestGridShot(opts *TestGridOptions) error {
+	logrus.Info("Starting krel testgrishot...")
+
+	if err := opts.Validate(); err != nil {
+		return errors.Wrap(err, "validating testgridshot options")
+	}
+
+	testgridJobs := []TestGridJob{}
+	for _, board := range opts.boards {
+		testGridDashboard := fmt.Sprintf("%s/sig-release-%s-%s/summary", opts.testgridURL, opts.branch, board)
+		content, err := http.GetURLResponse(testGridDashboard, false)
+		if err != nil {
+			return errors.Wrapf(err,
+				"unable to retrieve release announcement form url: %s", testGridDashboard,
+			)
+		}
+
+		var result map[string]interface{}
+		err = json.Unmarshal([]byte(content), &result)
+		if err != nil {
+			return errors.Wrap(err, "unable unmarshal the testgrid response")
+		}
+
+		testgridJobsTemp := []TestGridJob{}
+		for jobName, jobData := range result {
+			result := TestgridJobInfo{}
+			err = mapstructure.Decode(jobData, &result)
+			if err != nil {
+				return errors.Wrap(err, "failed to decode testgrid data")
```

Comment:
Just a nitty nit, but for wrapped errors I suggest that we remove words like "failed", because if we stack them we probably end up with something like: `failed to …: failed to …: failed to …` This seems noisy when trying to find the underlying cause of the error. I see those errors more as an additional level of context, this means we could do:
```suggestion
				return errors.Wrap(err, "decode testgrid data")
```